### PR TITLE
corsixth 0.65

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

**Comment by cho-m** [comment]
```
Error: corsixth: failed
An exception occurred within a child process:
  Minitest::Assertion: Expected [
...
"/opt/homebrew/opt/lua/lib/liblua.dylib", 
...
] to include "/opt/homebrew/opt/lua/lib/liblua.5.4.dylib".
```

**Comment by tobylane** [comment]
The release is tomorrow. I had a go at some more changes.

<details>
<summary>corsixth 0.65</summary>
<p>

```ruby
class Corsixth < Formula
  desc "Open source clone of Theme Hospital"
  homepage "https://github.com/CorsixTH/CorsixTH"
  url "https://github.com/CorsixTH/CorsixTH/archive/v0.65.tar.gz"
  sha256 "f951c9fcf7429accf035aeaeab681f7f9172af66b8475e9bacb32f5375515227"
  license "MIT"
  head "https://github.com/CorsixTH/CorsixTH.git"

  bottle do
    sha256 arm64_big_sur: "fb20eddb21a89396791d6ab3dfdf13c0bde91c44ba0a6f068b59194b361c4690"
    sha256 big_sur:       "9852913d485e6fce557001d16f78dac562b205c44810d93b133b539c02ed0436"
    sha256 catalina:      "bad3d139e3cac3c277a9bea632819fe27b90abfd7d5305813f839d78f5854ca6"
    sha256 mojave:        "a68aaf41d6feda1bbed25fa4fbb7ff73dc4b5049e23c13fb9377b22cb23c17a4"
  end

  depends_on "cmake" => :build
  depends_on "luarocks" => :build
  depends_on "ffmpeg"
  depends_on "freetype"
  depends_on "lua"
  depends_on "sdl2"
  depends_on "sdl2_mixer"

  def install
    args = std_cmake_args + %W[
      -DLUA_INCLUDE_DIR=#{Formula["lua"].opt_include}/lua
      -DLUA_LIBRARY=#{Formula["lua"].opt_lib}/liblua.dylib
      -DCORSIX_TH_DATADIR=#{prefix}/CorsixTH.app/Contents/Resources/
      -DWITH_LUAROCKS=ON
    ]
    system "cmake", ".", *args

    system "make"
    system "make", "install"
    bin.write_exec_script prefix/"CorsixTH.app/Contents/MacOS/CorsixTH"
  end

  test do
    app = prefix/"CorsixTH.app/Contents/MacOS/CorsixTH"
    assert_includes MachO::Tools.dylibs(app),
                    "#{lua.opt_lib}/liblua.#{Formula["lua"].version.major_minor}.dylib"
  end
end
```
</details>

This uses make install to have cmake pull the luarocks, avoiding the need for the env parts. I'm not sure what is the preferred way for this, but this simplifies the formula which is a good thing?  
I removed the Xcode line because it can use make not xcodebuild, not sure I understood this correctly.  
It's not portable but will be in the next release.

**Comment by carlocab** [comment]
> This uses make install to have cmake pull the luarocks, avoiding the need for the env parts. I'm not sure what is the preferred way for this, but this simplifies the formula which is a good thing?

We usually try to avoid build systems downloading things at build time, so I think we want to keep the existing `resource`s, which make it transparent in the formula that the extra resources are versioned and checksummed.

**Comment by carlocab** [comment]
Marking as do not merge since the release is tomorrow.

File: Formula/corsixth.rb [hunk 1]
```diff
@@ -73,10 +66,10 @@ def install
 
   test do
     # Make sure I point to the right version!
-    lua = Formula["lua@5.3"]
+    lua = Formula["lua"]
```
Comment (carlocab) [inline Formula/corsixth.rb:70]: We can remove these lines since they're not used.

Comment (chenrui333) [reply Formula/corsixth.rb]: yeah, just simplified the build a bit.

**Comment by chenrui333** [comment]
@tobylane thanks for the formula update suggestion. 👍

File: Formula/corsixth.rb [hunk 2]
```diff
@@ -16,64 +15,30 @@ class Corsixth < Formula
...
 
-    luapath = libexec/"vendor"
-    ENV["LUA_PATH"] = "#{luapath}/share/lua/5.3/?.lua"
-    ENV["LUA_CPATH"] = "#{luapath}/lib/lua/5.3/?.so"
+    args = std_cmake_args + %W[
+      -DLUA_INCLUDE_DIR=#{lua.opt_include}/lua
+      -DLUA_LIBRARY=#{lua.opt_lib}/liblua.dylib
+      -DCORSIX_TH_DATADIR=#{prefix}/CorsixTH.app/Contents/Resources/
+      -DWITH_LUAROCKS=ON
```
Comment (carlocab) [inline Formula/corsixth.rb:31]: I don't think we want this -- the resources makes it clearer that the extra downloads are versioned and checksummed.

**Review by carlocab — CHANGES_REQUESTED** [review]

File: Formula/corsixth.rb (hunk 2 above)
Comment (carlocab) [reply Formula/corsixth.rb:31]: It also looks like this option makes `corsixth` vendor its own Lua, which we don't want.

Comment (chenrui333) [reply Formula/corsixth.rb:31]: tackle on it now.

**Comment by chenrui333** [comment]
It looks like upstream has recalled the release, gonna close it now.

**Comment by tobylane** [comment]
Release is out, ready when you are. https://github.com/CorsixTH/CorsixTH/releases/tag/v0.65

**Comment by carlocab** [comment]
Opened #79628.

**Pull request closed without merging**